src/bson.c

```diff
@@ -31,6 +31,10 @@
 #include <Zend/zend_hash.h>
 #include <Zend/zend_interfaces.h>
 
+#if PHP_VERSION_ID >= 50400
+# include <Zend/zend_string.h>
+#endif
```

Comment:
I feel this should really be in the compat header as a #else condition for the str_efree() ?
Or is that pushing it a bit far?